### ISSUE TITLE
new args and kwargs options for publisher_acl are not documented at all and very little for external_auth

### ISSUE THREAD

**Issue by UtahDave** [saltstack/salt#42626]
`publisher_acl` and `external_auth` authentication have been improved upon to allow for args and kwargs to be whitelisted for functions.

The only documentation I could find was for the `external_auth` option here:  https://docs.saltstack.com/en/latest/topics/eauth/index.html#limiting-by-function-arguments

The docs aren't very clear on what exactly is being allowed and disallowed and the examples aren't very helpful.

I could find absolutely no docs on these options for `publisher_acl`,
https://docs.saltstack.com/en/latest/ref/publisheracl.html

but another user had spoken with a SaltStack PS engineer and has found the following to work:

```
publisher_acl:                                                                  
  user01:                                                                       
    - '*hdpmb*':                                                                
      - state.sls:                                                              
          args:                                                                 
          - 'config.linux.hadoop.*'                                             
      - test.ping                                                               
  user02:                                                                    
    - '*':                                                                      
      - test.ping                                                               
      - state.sls:                                                              
          args:                                                                 
          - 'config.linux.generic.motd'                                         
  user03:                                                                      
    - '*':                                                                      
      - test.ping                                                               
      - cmd.run:                                                                
          args:                                                                 
          - 'cat /etc/issue'                                                    
      - state.sls:                                                              
          args:                                                                 
          - 'config.linux.generic.motd'                                         
  user04:                                                                       
    - 'G@sid:*':                                                                
      - test.ping                                                               
      - grains.item                                                             
      - grains.items                                                            
      - state.sls:                                                              
          args:                                                                 
          - 'sap_basis.*'                                                       
      - state.show_sls:                                                         
          args:                                                                 
          - 'sap_basis.*'
```

Due to the sensitive nature of dealing with security, the documentation for these new options needs to be completed and filled out with examples. Preferably by the engineer who originally wrote it and understands what these options are supposed to be allowing

**Comment by UtahDave** [comment]
ZD 1619

**Comment by Ch3LL** [comment]
ping @DmitryKuzmenko can you take a look here for oxygen release? thanks!

**Comment by DmitryKuzmenko** [comment]
@Ch3LL I'll. Thank you.

**Comment by rallytime** [comment]
This is fixed with #45389
